File: 프로그래머스/0/181832. 정수를 나선형으로 배치하기/정수를 나선형으로 배치하기.java
class Solution {
    public int[][] solution(int n) {
        int[][] answer = new int[n][n];
        
        int left = 0;
        int right = n-1;
        int bottom = 0;
        int top = n-1;
        
        int num = 1;
        while(num<=n*n){
                for(int i = left ; i <= right; i++){
                    answer[left][i] = num;
                    num++;
                }
                bottom++;


                for(int i = bottom ; i <= top ; i++){
                    answer[i][top] = num;
                    num++;
                }
                top--;


                for(int i = right-1 ;i >= left ; i--){
                    answer[right][i] = num;
                    num++;
                }
                right--;


            for(int i = top ; i >= bottom; i--){
                answer[i][left] = num;
                num++;
            }
            left++;
        }   
        return answer;
    }
}

File: C#/프로그래머스/0/181832. 정수를 나선형으로 배치하기/정수를 나선형으로 배치하기.cs
using System;

public class Solution {
    public int[,] solution(int n) {
            int[,] answer = new int[n,n] ;

            int right = 0;
            int left = n - 1;
            int up = 0;
            int down = n - 1;

            int check = 1;
            while(check <= n * n)
            {
                for (int i = right; i <= left; i++)
                {                    
                    answer[up, i] = check++;
                }                
                up++;
                for(int i = up;i<=down;i++)
                {
                    answer[i, left] = check++;
                }
                left--;
                for (int i = left; i >= right; i--)
                    answer[down, i] = check++;
                down--;
                for (int i = down; i >= up; i--)
                    answer[i, right] = check++;
                right++;
            }

            for(int i = 0; i < answer.GetLength(0); i++)
            {
                for (int j = 0; j < answer.GetLength(1); j++)
                    Console.Write(answer[i, j] + " ");

                Console.WriteLine();
            }

            return answer;
    }
}